bootstrap/bootstrap.go

```diff
@@ -1737,16 +1771,40 @@ func (b *Bootstrap) writeBatchScript(cmd string) (string, error) {
 
 }
 
-func (b *Bootstrap) uploadArtifacts(ctx context.Context) error {
+func (b *Bootstrap) artifactPhase(ctx context.Context) error {
 	if b.AutomaticArtifactUploadPaths == "" {
 		return nil
 	}
 
-	span, ctx := tracetools.StartSpanFromContext(ctx, "upload artifacts", b.Config.TracingBackend)
+	spanName := b.implementationSpecificSpanName("phase/artifact", "artifact upload")
+	span, ctx := tracetools.StartSpanFromContext(ctx, spanName, b.Config.TracingBackend)
+	var err error
+	defer func() { span.FinishWithError(err) }()
+
+	err = b.preArtifactHooks(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = b.uploadArtifacts(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = b.postArtifactHooks(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run the pre-artifact hooks
+func (b *Bootstrap) preArtifactHooks(ctx context.Context) error {
+	span, ctx := tracetools.StartSpanFromContext(ctx, "pre-artifact", b.Config.TracingBackend)
 	var err error
 	defer func() { span.FinishWithError(err) }()
```

Comment:
The new spans for pre- and post-artifact hooks only add more information, and don't remove or change any existing information, so should be backwards compatible in datadog tracing.